runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java

```diff
@@ -133,7 +134,7 @@ public abstract class AbstractTransactionalContext implements
     /**
      * Cache of last known position of streams accessed in this transaction.
      */
-    protected final Map<UUID, Long> knownStreamsPosition = new HashMap<>();
+    protected final Map<ICorfuSMRProxyInternal<?>, Long> knownStreamsPosition = new HashMap<>();
```

Comment:
Would it be possible to use `VersionLockedObject` instead?

Reply:
I have reverted this back into a UUID (streamId).